buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -503,7 +504,10 @@ public ByteBuf getBytes(int index, ByteBuf dst, int length) {
 
     @Override
     public CharSequence getCharSequence(int index, int length, Charset charset) {
-        // TODO: We could optimize this for UTF8 and US_ASCII
+        if (CharsetUtil.US_ASCII.equals(charset) || CharsetUtil.ISO_8859_1.equals(charset)) {
```

Comment:
If you are going to include ISO_8859_1, I would like to see a unit test for this.  Particularly, if a String of every byte can be roundtripped.

Reply:
@carl-mastrangelo done!